internal/deployment/deployment_owner_registry.go

```diff
@@ -0,0 +1,47 @@
+package deployment
+
+import "sync"
+
+// Copyright (C) 2023 by Posit Software, PBC.
+type DeploymentOwnerRegistry struct {
+	sync.RWMutex
+	m map[string]string
+}
+
+var ActiveDeploymentRegistry DeploymentOwnerRegistry = DeploymentOwnerRegistry{
+	m: make(map[string]string),
+}
+
+func (o *DeploymentOwnerRegistry) Set(deploymentPath string, localId string) {
+	o.Lock()
+	defer o.Unlock()
+
+	if localId != "" {
+		o.m[deploymentPath] = localId
+	}
+}
+
+func (o *DeploymentOwnerRegistry) Check(deploymentPath string, localId string) bool {
+	o.Lock()
+	defer o.Unlock()
+
+	owner := o.m[deploymentPath]
+	return owner == localId
+}
+
+func (o *DeploymentOwnerRegistry) Clear(deploymentPath string, localId string) {
+	o.Lock()
+	defer o.Unlock()
+
+	owner := o.m[deploymentPath]
+	if owner == localId {
+		delete(o.m, deploymentPath)
+	}
```

Comment:
Nitpick: do we need to check if `o.m[deploymentPath]` exists?

> // The delete built-in function deletes the element with the specified key
// (m[key]) from the map. If m is nil or there is no such element, delete
// is a no-op.

Reply:
No, it was designed so that there must be a match of what localID the caller thinks is registered at the deploymentPath key, otherwise, it won't be deleted. This is to protect the staleness of information that parallel threads may have.

Reply:
> it was designed so that there must be a match of what localID the caller thinks is registered at the deploymentPath key, otherwise, it won't be deleted

Ah that makes sense now. Thanks for the explanation.